tidy3d/plugins/near2far/near2far.py

```diff
@@ -1,130 +1,483 @@
+        return ETA_0 / (n + 1j * k)
+
+    @classmethod
+    # pylint:disable=too-many-arguments
+    def from_surface_monitors(cls,
+        sim_data: SimulationData,
+        monitors: List[FieldMonitor],
+        normal_dirs: List[Direction],
+        frequency: float,
+        pts_per_wavelength: int = 10,
+        medium: Medium = None):
+        """Constructs :class:`Near2Far` from a list of surface monitors and their directions.
 
         Parameters
         ----------
         sim_data : :class:`.SimulationData`
-            Container for simulation data containing a near field monitor.
-        mon_name : str
-            Name of the :class:`.FieldMonitor` to use as source of near field.
-            Must be a :class:`.FieldMonitor` and stored in ``sim_data``.
+            Container for simulation data containing the near field monitors.
+        monitors : List[:class:`.FieldMonitor`]
+            List of :class:`.FieldMonitor` objects on which near fields will be sampled.
+        normal_dirs : List[:class:`.Direction`]
+            List containing the :class:`.Direction` of the normal to each surface monitor 
+            w.r.t. to the positive x, y or z unit vectors. Must have the same length as monitors.
         frequency : float
-            Frequency to select from the :class:`.FieldMonitor` to use for projection.
-            Must be a frequency stored in the :class:`FieldMonitor`.
+            Frequency to select from each :class:`.FieldMonitor` to use for projection.
+            Must be a frequency stored in each :class:`FieldMonitor`.
+        pts_per_wavelength : int
+            Number of points per wavelength with which to discretize the
+            surface monitors for the projection. Default: 10.
+        medium : :class:`.Medium`
+            Background medium in which to radiate near fields to far fields.
+            Default: same as the :class:`.Simulation` background medium.
+        """
+
+        if len(monitors) != len(normal_dirs):
+            raise SetupError("Number of monitors does not equal the number of directions.")
+
+        surfaces = []
+        for monitor, normal_dir in zip(monitors, normal_dirs):
+            surfaces.append(
+                Near2FarSurface(monitor=monitor, normal_dir=normal_dir)
+                )
+
+        return cls(
+            sim_data=sim_data,
+            surfaces=surfaces,
+            frequency=frequency,
+            pts_per_wavelength=pts_per_wavelength,
+            medium=medium
+            )
+
+    @pydantic.validator("currents", always=True)
+    def set_currents(cls, val, values):
+        """Sets the surface currents."""
+        sim_data = values.get("sim_data")
+        surfaces = values.get("surfaces")
+        pts_per_wavelength = values.get("pts_per_wavelength")
+        frequency = values.get("frequency")
+        medium = values.get("medium")
+        eps_complex = medium.eps_model(frequency)
+        n, _= medium.eps_complex_to_nk(eps_complex)
```

Comment:
so the background medium must be lossless?  If so, I wonder if we should grab `k` here and error if it's `!= 0`